sql/core/src/test/resources/sql-tests/inputs/datetime.sql

```diff
@@ -41,3 +41,21 @@ select timestamp'2019-10-06 10:11:12.345678' - date'2020-01-01';
 select 3 * (timestamp'2019-10-15 10:11:12.001002' - date'2019-10-15');
 select interval 4 month 2 weeks 3 microseconds * 1.5;
 select (timestamp'2019-10-15' - timestamp'2019-10-14') / 1.5;
+
+-- interval operation with null and zero case
+select interval '1 year 2 month' / null;
+select interval '1 year 2 month' / 0;
+select interval '4 months 2 weeks 6 days' * null;
+select null * interval '4 months 2 weeks 6 days';
```

Comment:
I think the above 4 tests are good enough as end-to-end tests. We should add UTs to make sure codgen and interpreted code path returns the same result.

Reply:
`select interval '1 year 2 month' / 0;`  shows the exact zero division issue with the rest one, I think we still need that to ensure correctness

Reply:
Yea we need that, we need the above 4 tests. Can we drop https://github.com/apache/spark/pull/26410/files#diff-e7a79cbcd5411e8891cee65c286c2177R51 and test codegen vs interpreted in UT?

Reply:
I mean this https://github.com/apache/spark/pull/26410/files#diff-e7a79cbcd5411e8891cee65c286c2177R58 is needed for comparing result with `select interval '1 year 2 month' / 0;`

Reply:
Can we pick a consistent style here? We can list all cases here by a) multiple `select interval_literal op null_or_zero`
or b) a single select.

Reply:
Well,
    https://github.com/apache/spark/blob/4c53ac18221cbd630bc16352e126d6fc56143239/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/IntervalExpressionsSuite.scala#L227

We already have a test case here which seems not able to verify this case. also cc @MaxGekk

Reply:
We are not facing the either-or thing here, it's the conflicting results by `a）` and `b）`

If the style here bothers you, we may change it like this, WDYT?
```sql
-- zero division
select interval '2 seconds' / 0;
select cast(k as interval) / v rom VALUES ('2 seconds', 0) t(k, v);

-- null division
select interval '2 seconds' / null;

-- null multiplication
select interval '2 seconds' * null;
select null * interval '2 seconds';
```